src/index.js

```diff
@@ -46,98 +59,77 @@ class PubsubPeerDiscovery extends Emittery {
    * after `this.delay` milliseconds
    */
   start () {
-    if (this._timeout) return
+    if (this._intervalId) return
 
     // Subscribe to pubsub
-    this.libp2p.pubsub.subscribe(TOPIC, (msg) => this._onMessage(msg))
+    for (const topic of this.topics) {
+      this.libp2p.pubsub.subscribe(topic, (msg) => this._onMessage(msg))
+    }
+
+    // Don't query if we are only listening
+    if (this._listenOnly) return
+
+    // Broadcast immediately, and then run on interval
+    this._broadcast()
+
     // Perform a delayed publish to give pubsub time to do its thing
-    this._timeout = setTimeout(() => {
-      this._query()
-    }, this.delay)
+    this._intervalId = setInterval(() => {
+      this._broadcast()
+    }, this.interval)
   }
 
   /**
    * Unsubscribes from the discovery topic
    */
   stop () {
-    clearTimeout(this._timeout)
-    this._timeout = null
-    this.libp2p.pubsub.unsubscribe(TOPIC)
+    clearInterval(this._intervalId)
+    this._intervalId = null
+    for (const topic of this.topics) {
+      this.libp2p.pubsub.unsubscribe(topic)
+    }
   }
 
   /**
-   * Performs a Query via Pubsub publish
+   * Performs a broadcast via Pubsub publish
    * @private
    */
-  _query () {
-    const id = randomBytes(32)
-    const query = {
-      id,
-      queryResponse: {
-        queryID: id,
-        publicKey: this.libp2p.peerInfo.id.pubKey.bytes,
-        addrs: this.libp2p.peerInfo.multiaddrs.toArray().map(ma => ma.buffer)
-      }
+  _broadcast () {
+    const peer = {
+      publicKey: this.libp2p.peerInfo.id.pubKey.bytes,
+      addrs: this.libp2p.peerInfo.multiaddrs.toArray().map(ma => ma.buffer)
+    }
+
+    const encodedPeer = PB.Peer.encode(peer)
+    for (const topic of this.topics) {
+      log('broadcasting our peer data on topic %s', topic)
+      this.libp2p.pubsub.publish(topic, encodedPeer)
     }
-    const encodedQuery = PB.Query.encode(query)
-    this.libp2p.pubsub.publish(TOPIC, encodedQuery)
   }
 
   /**
    * Handles incoming pubsub messages for our discovery topic
    * @private
    * @async
-   * @param {Message} message
+   * @param {Message} message A pubsub message
    */
   async _onMessage (message) {
-    if (await this._handleQuery(message.data)) return
-
-    this._handleQueryResponse(message.data)
-  }
-
-  /**
-   * Attempts to decode a QueryResponse from the given data. Any errors will be logged and ignored.
-   * @private
-   * @async
-   * @param {Buffer} data The Pubsub message data to decode
-   */
-  async _handleQuery (data) {
     try {
-      const query = PB.Query.decode(data)
-      const peerId = await PeerId.createFromPubKey(query.queryResponse.publicKey)
+      const peer = PB.Peer.decode(message.data)
+      const peerId = await PeerId.createFromPubKey(peer.publicKey)
+
```

Comment:
Should we add a log on received message? Might be useful for debugging

Reply:
Yes that would be helpful, it's now logging the peer id